test/AutoRest.CSharp.Tests/AutoRest.CSharp.Tests.csproj

```diff
@@ -0,0 +1,26 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```

Comment:
Do we need another test project?

Reply:
It definitely feels like a unit type of test compared to the existing 3 test projects, however that doesn't necessarily mean it needs to be in its own test project.

I will defer to you / chris on this

Reply:
We already do unit-testy stuff in https://github.com/Azure/autorest.csharp/blob/feature/v3/test/AutoRest.TestServer.Tests/Mgmt/ProviderSegmentDetectionTests.cs that's why I ask.

Reply:
@pakrym Thanks. I'll move the test case here.

But in the future, when there are more test cases focusing on generator itself (likely to happen after preview), we'd better think of putting them into a separate test projects, so that `AutoRest.TestServer.Tests` can focus on the validation of generated codes.